astropy/io/fits/file.py

```diff
@@ -83,6 +83,7 @@ class _File(object):
     @deprecated_renamed_argument('clobber', 'overwrite', '2.0')
     def __init__(self, fileobj=None, mode=None, memmap=None, overwrite=False,
                  cache=True):
+        from http.client import HTTPResponse
```

Comment:
While Python 2 compatibility is not dropped, it's better to use `from six.moves import http_client` instead, or we can defer to when Python 2 compatibility is dropped ? Not sure if this PR should or could easily be backported ?